### PR TITLE
Run tests with IPv6 address

### PR DESCRIPTION
**Pull request by natebosch**

### CODE REVIEW

**Comment by natebosch** [comment]
/sigh

Travis doesn't support IPv6, internal tests don't support IPv4 😢 

Will likely just disable the test internally.

**Comment by kevmoo** [comment]
You can do a sniff of Travis and switch to ipv4 – seems reasonable.

**Comment by kevmoo** [comment]
See https://docs.travis-ci.com/user/environment-variables/#Default-Environment-Variables

**Comment by natebosch** [comment]
Good idea!

updated PTAL

**Comment by natebosch** [comment]
Ugh, now the buildbots don't like using IPv6

https://luci-logdog.appspot.com/v/?s=chromium%2Fbb%2Fclient.dart.packages%2Fpackages-linux-shelf%2F1077%2F%2B%2Frecipes%2Fsteps%2Fpub_run_test%2F0%2Fstdout

File: test/io_server_test.dart [hunk 1]
```diff
@@ -16,7 +16,10 @@ import 'test_util.dart';
 void main() {
   var server;
   setUp(() async {
-    server = await IOServer.bind(InternetAddress.LOOPBACK_IP_V4, 0);
+    var address = Platform.environment.containsKey('TRAVIS')
+        ? InternetAddress.LOOPBACK_IP_V4
+        : InternetAddress.LOOPBACK_IP_V6;
```
Comment (nex3) [inline test/io_server_test.dart]: Can you add a comment explaining why we're testing on IPv6 on Travis and only Travis?

Comment (natebosch) [reply test/io_server_test.dart]: Sure. The chances that the `git blame` becomes at all difficult to find before the comment becomes a lie seems incredibly low but we can add one for now.

https://github.com/dart-lang/shelf/pull/89

**Pull request merged**